src/MobileDevice/widget/MobileDevice.ts

```diff
@@ -0,0 +1,126 @@
+    private onGetDeviceInformationMicroflow: string;
+    private onGetDeviceInformationOption: OnGetDeviceInformationOptions;
+    private onGetDeviceInformationPage?: string;
+    private mxObject: mendix.lib.MxObject;
+
+    update(mxObject: mendix.lib.MxObject, callback: () => void) {
+        this.mxObject = mxObject;
+        const warnings = this.validateProps();
+        if (warnings) {
+            domConstruct.create("div", {
+                class: "alert alert-danger",
+                innerHTML: warnings
+            }, this.domNode);
+        } else {
+            this.setUpWidgetDom();
+            this.connect(document, "deviceReady", dojoLang.hitch(this, this.onDeviceReady));
+            if (!window.device) {
+                mxObject.set(this.deviceIDAttribute, "");
+                mxObject.set(this.deviceTypeAttribute, "Web");
+                this.save(mxObject);
+            }
+        }
+
+        callback();
+    }
+
+    uninitialize(): boolean {
+        window.removeEventListener("deviceReady", this.onDeviceReady);
+        return true;
+    }
+
+    private setUpWidgetDom() {
+        domConstruct.create("div", {
+            class: "mobile-device-widget",
+            id: "mobile-device"
+        }, document.body, "first");
+    }
+
+    private onDeviceReady() {
+        if (cordova.getAppVersion) {
+            cordova.getAppVersion.getVersionNumber((versionNumber: number) => {
+                if (versionNumber) {
+                    cordova.getAppVersion.getVersionCode((versionCode: string) => {
+                        if (versionCode) {
+                            this.mxObject.set(this.deviceIDAttribute, window.device.uuid);
+                            this.mxObject.set(this.deviceTypeAttribute, window.device.platform);
+                            this.mxObject.set(this.appVersionBuildAttribute, versionNumber);
+                            this.mxObject.set(this.appVersionVersionAttribute, versionCode);
+                            this.save(this.mxObject);
+                        }
+                    });
+                }
+            });
+        } else {
+            window.mx.ui.error("Add cordova-plugin-app-version to your project config");
+        }
+    }
+
+    private validateProps(): string {
+        let errorMessage = "";
+        if (this.onGetDeviceInformationOption === "callMicroflow" && !this.onGetDeviceInformationMicroflow) {
+            errorMessage = "on click microflow is required in the 'Events' tab, 'Microflow' property";
+        } else if (this.onGetDeviceInformationOption === "showPage" && !this.onGetDeviceInformationPage) {
+            errorMessage = "on click page is required in the 'Events' tab, 'Page' property";
+        }
+
+        return errorMessage && `Error in device id widget configuration: ${errorMessage}`;
+    }
+
+    private save(mxObject: mendix.lib.MxObject) {
+        window.mx.data.commit({
+            callback: () => {
+                mx.ui.back();
```

Comment:
`mx.ui.back();` Should this always happen? Check with Danny, and document if needed.